Doc/library/io.rst

```diff
@@ -911,21 +923,22 @@ Text I/O
 
    .. method:: seek(offset, whence=SEEK_SET, /)
 
-      Change the stream position to the given *offset*.  Behaviour depends on
-      the *whence* parameter.  The default value for *whence* is
-      :data:`SEEK_SET`.
-
-      * :data:`SEEK_SET` or ``0``: seek from the start of the stream
-        (the default); *offset* must either be a number returned by
-        :meth:`TextIOBase.tell`, or zero.  Any other *offset* value
-        produces undefined behaviour.
-      * :data:`SEEK_CUR` or ``1``: "seek" to the current position;
-        *offset* must be zero, which is a no-operation (all other values
-        are unsupported).
-      * :data:`SEEK_END` or ``2``: seek to the end of the stream;
-        *offset* must be zero (all other values are unsupported).
-
-      Return the new absolute position as an opaque number.
+      Change the stream position to the given byte *offset*,
+      and return the new absolute position as an opaque number.
+      *offset* is interpreted relative to the position indicated by *whence*.
+      Valid values for *whence* are:
+
+      * :data:`SEEK_SET` -- seek from the start of the stream (the default);
+        *offset* must either be a number returned by
+        :meth:`TextIOBase.tell`, or zero.
```

Comment:
It would be better to describe what will happen exactly.
```suggestion
        *offset* must either be a number returned by
        :meth:`TextIOBase.tell`, in which case the stream is returned to the position before calling :meth:`!tell()`, or zero, in which case the stream is rewound to the start.
```